nexus/src/db/datastore.rs

```diff
@@ -1607,19 +1625,24 @@ impl DataStore {
                 insert_new_query.insert_and_get_results(conn).map_err(|e| {
                     match e {
                         SyncInsertError::CollectionNotFound => {
-                            diesel::result::Error::RollbackTransaction
+                            UpdateFirewallRulesError::CollectionNotFound
                         }
-                        SyncInsertError::DatabaseError(e) => e,
+                        SyncInsertError::DatabaseError(e) => e.into(),
                     }
                 })
             })
             .await
-            .map_err(|e| {
-                public_error_from_diesel_pool(
-                    e,
-                    ResourceType::VpcFirewallRule,
-                    LookupType::ById(*vpc_id),
-                )
+            .map_err(|e| match e {
+                UpdateFirewallRulesError::CollectionNotFound => {
+                    Error::not_found_by_id(ResourceType::Vpc, vpc_id)
+                }
+                UpdateFirewallRulesError::Pool(e) => {
+                    public_error_from_diesel_pool(
+                        e,
+                        ResourceType::VpcFirewallRule,
+                        LookupType::ById(*vpc_id),
```

Comment:
It's *kinda* unrelated to this PR, but I'm not actually sure this is doing the right thing. I'm pretty sure would bottom out here: https://github.com/oxidecomputer/omicron/blob/35af1bdf1a3d4f1d3c7f5e84e0edbe8aa394e9bd/nexus/src/db/error.rs#L129-L132

Which would return an error saying "We can't find a VPC firewall rule with the VPC ID". But that's not exactly right - although, we may have failed to perform an operation on a VPC Firewall Rule, each firewall rule would have their own IDs, not the VPC ID, right?

Reply:
Yeah, though right now all  API actions are in terms of the VPC, so we never get lookups by firewall rule ID. I think it should probably be ResourceType::Vpc though